cdc/cdc.cc

```diff
@@ -323,6 +329,75 @@ class transformer final {
         }
         return res;
     }
+
+    static db::timeout_clock::time_point default_timeout() {
+        return db::timeout_clock::now() + 10s;
+    }
+
+    future<lw_shared_ptr<cql3::untyped_result_set>> pre_image_select(
+            service::storage_proxy& proxy,
+            service::client_state& client_state,
+            db::consistency_level cl,
+            const mutation& m)
+    {
+        auto& p = m.partition();
+        if (p.partition_tombstone() || !p.row_tombstones().empty() || p.clustered_rows().empty()) {
+            return make_ready_future<lw_shared_ptr<cql3::untyped_result_set>>();
+        }
+
+        auto& pk = m.key();
+
+        dht::partition_range_vector partition_ranges{dht::partition_range(dht::global_partitioner().decorate_key(*_schema, pk))};
```

Comment:
how about using m.decorated_key() here instead of decorating the key again?

Reply:
Indeed.